### PR TITLE
BF: Python-apt working under Travis & support for attrs 17.1.0

### PR DESCRIPTION
**Pull request by rbuccigrossi**
Clearly the purpose of this branch changed. After figuring out what was different between our environments and travis' docker images, I finally got python-apt working (and a quick fix to support attrs 17.1.0)

### CODE REVIEW

**Comment by yarikoptic** [comment]
Might be worth keeping ultimate discovery .travis.yml handy somewhere under tools/ci so we could just copy it when needed to gather info

**Comment by rbuccigrossi** [comment]
@yarikoptic Please take a look. I think I finally have python-apt working on travis' docker images!

File: .travis.yml [hunk 1]
```diff
@@ -51,7 +51,7 @@ before_install:
 
 install:
   # all mess below to get an elderly python-apt installed on elderly precise within python env -- we must build it
-  - sudo tools/ci/build_install_apt
+  - bash tools/ci/build_install_apt
```
Comment (yarikoptic) [inline .travis.yml]: FWIW -- there should be no need for explicit `bash` here -- file has shebang and is executable

File: niceman/retrace/packagemanagers.py [hunk 2]
```diff
@@ -153,7 +153,7 @@ def _create_package(self, pkgname):
         raise NotImplementedError
 
 
-@attr.s
+@attr.s(cmp=True, hash=True, frozen=True)
```
Comment (yarikoptic) [inline niceman/retrace/packagemanagers.py]: from which version of apt it is ?  adjust setup.py accordingly please

File: niceman/retrace/packagemanagers.py [hunk 3]
```diff
@@ -215,8 +215,16 @@ def _create_named_origin(o, used_names):
...
+        new_o = DpkgOrigin(name=name,
+                           component=o.component,
+                           codename=o.codename,
+                           archive=o.archive,
+                           architecture=o.architecture,
+                           origin=o.origin,
+                           label=o.label,
+                           site=o.site,
+                           archive_uri=o.archive_uri,
+                           date=o.date)
```
Comment (yarikoptic) [inline niceman/retrace/packagemanagers.py]: there should be a better way ;)  e.g. if you go through all _attrs, compile a dict (e.g. params) with adjusted name and pass into `DpkgOrigin(**params)`

File: niceman/retrace/tests/test_packagemanagers.py [hunk 4]
```diff
@@ -80,7 +80,7 @@ def test_dpkg_manager_identify_packages():
             assert o.label, "A non-local origin needs a label"
             assert o.site, "A non-local origin needs a site"
             assert o.archive_uri, "An archive_uri should have been found"
-            assert o.date, "An package should have been found"
```
Comment (yarikoptic) [inline niceman/retrace/tests/test_packagemanagers.py]: irrelevant to this PR but "duplication is evil".  use `.__attrs_attrs__` and `getattr` to get them in a loop ;)

File: niceman/retrace/packagemanagers.py (hunk 2 above)
Comment (rbuccigrossi) [inline niceman/retrace/packagemanagers.py]: Frozen was introduced in 16.1.0 of attrs, so I'll add that dependency,

File: .travis.yml (hunk 1 above)
Comment (rbuccigrossi) [inline .travis.yml]: Hey, look at that: git kindly maintains executable status! Nice to know.

File: niceman/retrace/tests/test_packagemanagers.py (hunk 4 above)
Comment (rbuccigrossi) [inline niceman/retrace/tests/test_packagemanagers.py]: Well, I still want to explicitly list the mandatory attributes, so let's see if you like the result.

Comment (yarikoptic) [inline niceman/retrace/tests/test_packagemanagers.py]: sure... then
```python
for mandatory_attribute in ['date', ...]:
    assert getattr(o, mandatory_attribute), "Value for %s is missing" % mandatory_attribute
```

**Pull request merged**